docs/source/design/ui_design.rst

```diff
@@ -0,0 +1,328 @@
+
+PSI4
+====
+PSI4 is a C++/Python (70%/30%) open-source quantum chemistry package distributed
+under the LGPL3 license. PSI4 provides two different types of UI referred to as
+Psithon and PsiAPI modes. In the Psithon mode, the user writes an input file in
+a domain specific language similar to Python. In the PsiAPI mode, the user can
+write a pure Python script that interacts with PSI4 as a Python module. Since
+the latter is more relevant to our design, we show below how to run an SCF
+calculation for a hydrogen molecule using the PsiAPI mode.
+
+.. code-block:: python
+
+    import psi4
+    mol = psi4.geometry('H 0. 0. 0. \n H 0. 0. 1.')
+    energy= psi4.energy('scf/sto-3g')    
+
+Here, mol is the molecule object, which is created using the ``psi4.geometry()``
+function and the energy is computed using the ``psi4.energy()`` function, which
+takes the method name as the required argument. Note that the user do not need
+to pass the molecule object explicitly to the subsequent energy calculation. By
+default energy is computed for the last molecule defined with the
+``psi4.geometry()`` function. Additional options can be set with the
+``psi4.set_options()`` function, which takes a Python dictionary as the required
+argument. 
+
+QCEngine
+========
+QCEngine is a general purpose quantum chemistry program interface. It is a
+Python library that provides a common API for quantum chemistry programs.
+QCEngine is an open-source package distributed under the Apache License 2.0.
+
+Below you can find how to run an SCF calculation for a hydrogen molecule using
+QCEngine.
+
+.. code-block:: python
+
+    import qcengine as qcng
+    import qcelemental as qcel
+    mol = qcel.models.Molecule.from_data('H 0. 0. 0. \n H 0. 0. 1.')
+    out = qcng.compute({"molecule": mol, "driver": "energy", "model": {"method":
+    "SCF", "basis": "sto-3g"}}, "NWChemEx")
+    energy = out.return_result
+
+Here, mol is the molecule object (type
+``qcelemental.models.molecule.Molecule``), which is created using the
+``qcel.models.Molecule.from_data()`` function from QCElemental package. The SCF
+energy is computed using the ``qcng.compute()`` function, from QCEngine
+package. Note that, the input for the ``qcng.compute`` function is a Python
+dictionary with a schema defined by QCElemental.
+
+NWChemEx
+========
+
+In NWChemEx, running an SCF calculation for a hydrogen molecule is made easy
+through the use of a Python function with named arguments, as shown below.
+
+.. code-block:: python
+
+    import nwchemex as nwx 
+    energy = nwx.calculate_scf_energy(molecule = 'H 0. 0. 0. \n H 0. 0. 1.', basis = 'sto-3g')
+
+While the example shows only the two required arguments for the ``nwx.calculate_scf_energy()``
+function, the whole function signature is given below: 
+
+.. code-block:: python
+
+    def calculate_scf_energy(molecule: Union[str, chemist.Molecule], basis:
+    Union[str, simde.type.ao_space], spin: int = 1, max_iterations: int = 50,
```

Comment:
spin is part of the molecule.

Reply:
Although it is part of the molecule, a user may specify the molecule with an xyz string without the spin info, which I think will be a common scenario. In this case, we will use the spin given in the options. If the user provides the `chemist.Molecule` with a spin different than `options.spin`, we should pick the higher one and print a warning about the spin used in the calculation and if molecule object is modified.

Reply:
By convention, XYZ format usually starts with charge and multiplicity on their own line, followed by an optional comment line, followed by the atoms and coordinates. If the user does not put the charge and multiplicity in the string I'd suggest we default to neutral/singlet (and the `Molecule` class should throw if that doesn't make sense) rather than introducing a new option for it. 

As you mentioned, introducing a separate mechanism for spin leads to potential ambiguity arising from two potential sources of truth. When possible, my vote is always for a single source of truth.